test/Kestrel.FunctionalTests/ResponseTests.cs

```diff
@@ -2471,46 +2471,40 @@ public Task ResponseStatusCodeSetBeforeHttpContextDisposedRequestMalformedReadIg
                 }
             };
 
-            var aborted = new ManualResetEventSlim();
-
             using (var server = new TestServer(async context =>
             {
-                context.RequestAborted.Register(() =>
-                {
-                    aborted.Set();
-                });
+                context.RequestAborted.Register(() => requestAborted.Set());
 
-                context.Response.ContentLength = chunks * chunkSize;
+                context.Response.ContentLength = responseSize;
 
                 for (var i = 0; i < chunks; i++)
                 {
                     await context.Response.WriteAsync(new string('a', chunkSize), context.RequestAborted);
                 }
             }, testContext))
             {
-                using (var connection = server.CreateConnection())
+                using (var socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
                 {
-                    await connection.Send(
-                        "GET / HTTP/1.1",
-                        "Host:",
-                        "",
-                        "");
+                    socket.ReceiveBufferSize = 1;
 
-                    Assert.True(aborted.Wait(TimeSpan.FromSeconds(60)));
+                    socket.Connect(new IPEndPoint(IPAddress.Loopback, server.Port));
+                    socket.Send(Encoding.ASCII.GetBytes("GET / HTTP/1.1\r\nHost: \r\n\r\n"));
 
-                    await connection.Receive(
-                        "HTTP/1.1 200 OK",
-                        "");
-                    await connection.ReceiveStartsWith("Date: ");
-                    await connection.Receive(
-                        $"Content-Length: {chunks * chunkSize}",
-                        "",
-                        "");
+                    Assert.True(messageLogged.Wait(TimeSpan.FromSeconds(60)));
+                    Assert.True(requestAborted.Wait(TimeSpan.FromSeconds(60)));
 
-                    await Assert.ThrowsAsync<EqualException>(async () => await connection.ReceiveForcedEnd(
-                        new string('a', chunks * chunkSize)));
+                    var buffer = new byte[chunkSize];
+                    var totalReceived = 0;
+                    var received = socket.Receive(buffer);
+
+                    while (received > 0)
+                    {
+                        totalReceived += received;
+                        received = socket.Receive(buffer);
```

Comment:
Maybe put this in a try/catch and break out of the loop in the case Receive throws an IOException.